CDP4JsonFileDal/JsonFileDalUtils.cs

```diff
@@ -152,23 +152,70 @@ internal static void AddDomainsOfExpertise(EngineeringModelSetup engineeringMode
         }
 
         /// <summary>
-        /// Adds the <see cref="Person"/> of the <see cref="Participant"/> of the <see cref="engineeringModelSetup"/>
-        /// to the target <see cref="HashSet{Person}"/> if this does not already contain them
+        /// Add <see cref="Person"/> and related objects: <see cref="PersonRole"/>, <see cref="ParticipantRole"/>, <see cref="Organization"/> to a specific hash set passed by reference
         /// </summary>
         /// <param name="engineeringModelSetup">
         /// The <see cref="EngineeringModelSetup"/> that contains the active <see cref="DomainOfExpertise"/> instances.
         /// </param>
         /// <param name="persons">
         /// The target <see cref="HashSet{Person}"/>
         /// </param>
-        internal static void AddPersons(EngineeringModelSetup engineeringModelSetup, ref HashSet<Person> persons)
+        /// <param name="personRoles">
+        /// The target <see cref="HashSet{PersonRole}"/>
+        /// </param>
+        /// <param name="participantRoles">
+        /// The target <see cref="HashSet{ParticipantRole}"/>
+        /// </param>
+        /// <param name="organizations">
+        /// The target <see cref="HashSet{Organization}"/>
+        /// </param>///
+        internal static void AddPersons(EngineeringModelSetup engineeringModelSetup, ref HashSet<Person> persons, ref HashSet<PersonRole> personRoles, ref HashSet<ParticipantRole> participantRoles, ref HashSet<Organization> organizations)
         {
             foreach (var participant in engineeringModelSetup.Participant)
             {
                 if (!persons.Contains(participant.Person))
```

Comment:
I would say that the `if` statements are not necessary for a hashset (assuming the `Thing` objects contain overrides for the necessary methods (`equals` and `hashCode` or their equivalents).

Reply:
Correct. A HashSet it's a collection that contains no duplicate elements that are in no particular order. And if you try to add a new one that already exists the HashSet will not be modified.

Reply:
Presuming they respect the `Equals` and `GetHashCode` contract.